react-native-spokestack.d.ts

```diff
@@ -0,0 +1,327 @@
+export = RNSpokestack;
+export as namespace RNSpokestack;
+
+declare namespace RNSpokestack {
+  enum TraceLevel {
+    DEBUG = 10,
+    PERF = 20,
+    INFO = 30,
+    NONE = 100,
+  }
+
+  enum TTSFormat {
+    TEXT = 0,
+    SSML = 1,
+    SPEECHMARKDOWN = 2,
+  }
+
+  interface SpokestackRecognizeEvent {
+    transcript?: string;
+  }
+
+  interface SpokestackErrorEvent {
+    error?: string;
+  }
+
+  interface SpokestackTraceEvent {
+    message?: string;
+  }
+
+  interface SpokestackTTSEvent {
+    url?: string;
+  }
+
+  interface NLUSlot {
+    type: string;
+    value: string;
+  }
+
+  interface SpokestackNLUEvent {
+    result?: {
+      intent: string;
+      confidence: number;
+      slots: NLUSlot[];
+    };
+  }
+
+  interface SynthesizeOptions {
+    input: string;
+    format: TTSFormat;
+    voice: string;
+  }
+
+  type SpokestackEvent = SpokestackRecognizeEvent &
+    SpokestackErrorEvent &
+    SpokestackTraceEvent &
+    SpokestackTTSEvent &
+    SpokestackNLUEvent;
+
+  type SpokestackInput =
+    | "io.spokestack.spokestack.android.MicrophoneInput"
+    | "io.spokestack.spokestack.android.NoInput"
+    | "io.spokestack.spokestack.android.PreASRMicrophoneInput";
+
+  type SpokestackVoiceActivity =
+    | "io.spokestack.spokestack.webrtc.VoiceActivityDetector"
+    | "io.spokestack.spokestack.webrtc.VoiceActivityTrigger"
+    // Specify the two above, the one below, or none
+    | "io.spokestack.spokestack.wakeword.WakewordTrigger";
+
+  // Only specify one of these
+  type SpokestackSpeechRecognizer =
+    | "io.spokestack.spokestack.android.AndroidSpeechRecognizer"
+    | "io.spokestack.spokestack.google.GoogleSpeechRecognizer"
+    | "io.spokestack.spokestack.microsoft.AzureSpeechRecognizer";
+
+  // Important: follow this order when specifying stages
+  type SpokestackStage =
+    | "io.spokestack.spokestack.webrtc.AcousticNoiseSuppressor"
+    | "io.spokestack.spokestack.webrtc.AutomaticGainControl"
+    | SpokestackVoiceActivity
+    | SpokestackSpeechRecognizer
+    | "io.spokestack.spokestack.ActivationTimeout";
```

Comment:
These pieces are all optional, you can have one or none of each. Eg just VAD, or just a speech recognizer.